sdk/keyvault/azure_security_keyvault_secrets/README.md

```diff
@@ -1,24 +1,419 @@
+    let secret: SecretBundle = client
+        .set_secret(
+            "secret-name".into(),
+            secret_set_parameters.try_into()?,
+            None,
+        )
+        .await?
+        .into_body()
+        .await?;
+
+    println!(
+        "Secret Name: {:?}, Value: {:?}, Version: {:?}",
+        secret.resource_id()?.name,
+        secret.value,
+        secret.resource_id()?.version
+    );
+
+    Ok(())
+}
+```
+
+### Retrieve a secret
+
+`get_secret` retrieves a secret previously stored in the Azure Key Vault. Setting the `secret-version` to an empty string will return the latest version.
+
+```rust no_run
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{models::SecretBundle, SecretClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Retrieve a secret using the secret client.
+    let secret: SecretBundle = client
+        .get_secret("secret-name".into(), "secret-version".into(), None)
+        .await?
+        .into_body()
+        .await?;
+
+    println!("Secret Value: {:?}", secret.value);
+
+    Ok(())
+}
+```
+
+### Update an existing secret
+
+`update_secret` updates a secret previously stored in the Azure Key Vault. Only the attributes of the secret are updated. To update the value, call `SecretClient::set_secret` on a secret with the same name.
+
+```rust no_run
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{
+    models::{SecretAttributes, SecretUpdateParameters},
+    SecretClient,
+};
+use std::collections::HashMap;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Update a secret using the secret client.
+    let secret_update_parameters = SecretUpdateParameters {
+        content_type: Some("text/plain".into()),
+        secret_attributes: Some(SecretAttributes::default()),
+        tags: Some(HashMap::from_iter(vec![(
+            "tag-name".into(),
+            "tag-value".into(),
+        )])),
+    };
+
+    client
+        .update_secret(
+            "secret-name".into(),
+            "".into(),
+            secret_update_parameters.try_into()?,
+            None,
+        )
+        .await?
+        .into_body()
+        .await?;
+
+    Ok(())
+}
+```
+
+### Delete a secret
+
+`delete_secret` starts a long-running operation to delete a secret previously stored in the Azure Key Vault. You can retrieve the secret immediately without waiting for the operation to complete. When [soft-delete] is not enabled for the Azure Key Vault, this operation permanently deletes the secret.
```

Comment:
1. Soft delete is enabled by default and cannot be disabled anymore.
2. Mention of an LRO might make people think we have client support but won't.

So maybe instead,

```suggestion
`delete_secret` will tell Key Vault to delete a secret but it is not deleted immediately. It is not deleted until the service-configured data retention period - the default is 90 days - or until you call `purge_secret` on the returned `DeletedSecretBundle.id`.
```

Reply:
Added suggestion